diff --git a/tools/makesis-4/src/signutils.cpp b/tools/makesis-4/src/signutils.cpp
index 68838c4..f085ec6 100644
--- a/tools/makesis-4/src/signutils.cpp
+++ b/tools/makesis-4/src/signutils.cpp
@@ -113,7 +113,7 @@ SISSignature* makeSignature(SISField* controller, const char* keyData, int keyLe
 	BIO_free_all(io);
 
 	if (type == SigAuto) {
-		if (key->type == EVP_PKEY_DSA || key->type == EVP_PKEY_DSA1 || key->type == EVP_PKEY_DSA2 || key->type == EVP_PKEY_DSA3 || key->type == EVP_PKEY_DSA4)
+		if (EVP_PKEY_base_id(key) == EVP_PKEY_DSA)
 			type = SigDsa;
 		else
 			type = SigRsa;
@@ -123,39 +123,38 @@ SISSignature* makeSignature(SISField* controller, const char* keyData, int keyLe
 	uint8_t* ptr = buffer;
 	controller->CopyFieldData(ptr);
 
-	EVP_MD_CTX ctx, verify;
-	EVP_MD_CTX_init(&ctx);
-	EVP_MD_CTX_init(&verify);
-	const EVP_MD* md = NULL;
-	if (type == SigDsa)
-		md = EVP_dss1();
-	else
-		md = EVP_sha1();
-	if (EVP_SignInit_ex(&ctx, md, NULL) == 0) {
+	EVP_MD_CTX* ctx = EVP_MD_CTX_new();
+	EVP_MD_CTX* verify = EVP_MD_CTX_new();
+	if (!ctx || !verify) {
+		ERR_print_errors_fp(stderr);
+		throw SignOpenSSLErr;
+	}
+	const EVP_MD* md = EVP_sha1();
+	if (EVP_SignInit_ex(ctx, md, NULL) == 0) {
 		ERR_print_errors_fp(stderr);
 		throw SignOpenSSLErr;
 	}
-	if (EVP_VerifyInit(&verify, md) < 0) {
+	if (EVP_VerifyInit(verify, md) < 0) {
 		ERR_print_errors_fp(stderr);
 		throw SignOpenSSLErr;
 	}
-	if (EVP_SignUpdate(&ctx, buffer, controller->Length()) == 0) {
+	if (EVP_SignUpdate(ctx, buffer, controller->Length()) == 0) {
 		ERR_print_errors_fp(stderr);
 		throw SignOpenSSLErr;
 	}
-	if (EVP_VerifyUpdate(&verify, buffer, controller->Length()) == 0) {
+	if (EVP_VerifyUpdate(verify, buffer, controller->Length()) == 0) {
 		ERR_print_errors_fp(stderr);
 		throw SignOpenSSLErr;
 	}
 	delete [] buffer;
 	unsigned int siglen = EVP_PKEY_size(key);
 	uint8_t* signature = new uint8_t[siglen];
-	if (EVP_SignFinal(&ctx, signature, &siglen, key) == 0) {
+	if (EVP_SignFinal(ctx, signature, &siglen, key) == 0) {
 		ERR_print_errors_fp(stderr);
 		throw SignOpenSSLErr;
 	}
-	EVP_MD_CTX_cleanup(&ctx);
-	int ret = EVP_VerifyFinal(&verify, signature, siglen, publicKey);
+	EVP_MD_CTX_free(ctx);
+	int ret = EVP_VerifyFinal(verify, signature, siglen, publicKey);
 	if (ret < 0) {
 		ERR_print_errors_fp(stderr);
 		throw SignOpenSSLErr;
@@ -164,7 +163,7 @@ SISSignature* makeSignature(SISField* controller, const char* keyData, int keyLe
 		fprintf(stderr, "Could not verify signature with certificate\n");
 		throw SignBadKey;
 	}
-	EVP_MD_CTX_cleanup(&verify);
+	EVP_MD_CTX_free(verify);
 	EVP_PKEY_free(key);
 
 
@@ -274,9 +273,6 @@ SISCertificateChain* makeChain(const char* certData, int certLen, EVP_PKEY** pub
 			*publicKey = X509_PUBKEY_get(X509_get_X509_PUBKEY(cert));
 		i2d_X509_bio(out, cert);
 
-		X509_OBJECT obj;
-		obj.type = X509_LU_X509;
-		obj.data.x509 = cert;
-		X509_OBJECT_free_contents(&obj);
+		X509_free(cert);
 	}
 	BIO_free_all(in);
@@ -309,9 +305,6 @@ SISCertificateChain* makeChain(const char* certData, int certLen, EVP_PKEY** pub
 			*publicKey = X509_PUBKEY_get(X509_get_X509_PUBKEY(cert));
 		i2d_X509_bio(out, cert);
 
-		X509_OBJECT obj;
-		obj.type = X509_LU_X509;
-		obj.data.x509 = cert;
-		X509_OBJECT_free_contents(&obj);
+		X509_free(cert);
 	}
 	BIO_free_all(in);
diff --git a/tools/makesis-4/src/signsis.cpp b/tools/makesis-4/src/signsis.cpp
index e7e1a6a..da06192 100644
--- a/tools/makesis-4/src/signsis.cpp
+++ b/tools/makesis-4/src/signsis.cpp
@@ -248,10 +248,7 @@ int main(int argc, char *argv[]) {
 				else
 					displayCert(cert);
 
-				X509_OBJECT obj;
-				obj.type = X509_LU_X509;
-				obj.data.x509 = cert;
-				X509_OBJECT_free_contents(&obj);
+				X509_free(cert);
 			}
 
 			if (out)
