diff --git a/libquickjs-sys/embed/quickjs/quickjs.c b/libquickjs-sys/embed/quickjs/quickjs.c
index 39e334c..92a4ecf 100644
--- a/libquickjs-sys/embed/quickjs/quickjs.c
+++ b/libquickjs-sys/embed/quickjs/quickjs.c
@@ -34,6 +34,14 @@
 #include <time.h>
 #include <fenv.h>
 #include <math.h>
+#ifdef __SYMBIAN32__
+#define alloca __builtin_alloca
+static double qjs_symbian_log2(double x)
+{
+    return log(x) / 0.69314718055994530942;
+}
+#define log2 qjs_symbian_log2
+#endif
 #ifdef __PSP__
 #ifndef FE_TONEAREST
 #define FE_TONEAREST 0
@@ -97,7 +105,7 @@ _Static_assert(_Alignof(JSValue) == 8, "Vita JSValue must be 8-byte aligned");
 #if !defined(__EMSCRIPTEN__)
 #define CONFIG_ATOMICS
 #endif
-#if defined(__PSP__) || defined(__vita__)
+#if defined(__PSP__) || defined(__vita__) || defined(__SYMBIAN32__)
 #undef CONFIG_ATOMICS
 #endif
 
@@ -2173,6 +2181,8 @@ static size_t js_def_malloc_usable_size(const void *ptr)
     return 0;
 #elif defined(__vita__)
     return malloc_usable_size((void *)ptr);
+#elif defined(__SYMBIAN32__)
+    return 0;
 #elif defined(__linux__) || defined(__GLIBC__)
     return malloc_usable_size((void *)ptr);
 #else
@@ -47302,7 +47312,7 @@ static int getTimezoneOffset(int64_t time)
         res = (gm_ti - loc_ti) / 60;
     }
 #else
-#if defined(__PSP__) || defined(__vita__)
+#if defined(__PSP__) || defined(__vita__) || defined(__SYMBIAN32__)
     (void)ti;
     res = 0;
 #else
diff --git a/libquickjs-sys/embed/quickjs/quickjs.h b/libquickjs-sys/embed/quickjs/quickjs.h
index ff6c948..31a2b61 100644
--- a/libquickjs-sys/embed/quickjs/quickjs.h
+++ b/libquickjs-sys/embed/quickjs/quickjs.h
@@ -1065,7 +1065,8 @@ static inline JSValue JS_NewCFunctionMagic(JSContext *ctx, JSCFunctionMagic *fun
                                            int length, JSCFunctionEnum cproto, int magic)
 {
     /* Used to squelch a -Wcast-function-type warning. */
-    JSCFunctionType ft = { .generic_magic = func };
+    JSCFunctionType ft;
+    ft.generic_magic = func;
     return JS_NewCFunction2(ctx, ft.generic, name, length, cproto, magic);
 }
 int JS_SetConstructor(JSContext *ctx, JSValueConst func_obj,
